File: Module-06/01-colonnes-df/05-agregations.scala
import org.apache.spark.sql.functions._

meteo.where("id < 8000").
     select("annee","mois_jour",
             "temperature").
     describe().show()

meteo.where("id < 8000").count()
meteo.count()

meteo.where("id < 8000").
     select("humidite","visibilite","pression").
     describe().show()


meteo.where("id < 8000").
          groupBy("id","annee").
          agg(
                 count("id").alias("nb_villes"),
                 round(avg("temperature"),2).alias("temperature"),
                 round(avg("humidite"),2).alias("humidite"),
                 round(avg("visibilite"),2).alias("visibilite"),
                 round(avg("pression"),2).alias("pression")
          ).show(10)

meteo.where("id < 8000").
               groupBy("id","annee").
               agg(
                      "id"->"count",
                      "temperature"->"avg",
                      "humidite"->"avg"
               ).toDF("id","annee","nb_villes","temperature",
                      "humidite").show(10)

meteo.where("id < 8000").
     groupBy("id","mois").
     agg(round(avg("temperature"),2)).
     orderBy("id","mois").
     show()

meteo.where("id < 8000").
    groupBy("id").
    pivot("mois").
    agg(round(avg("temperature"),2)).
    toDF("id","Jan","Fev","Mar","Avr","Mai",
         "Jun","Jul","Aou","Sep","Oct","Nov","Dec").
    orderBy("id").
    show()

File: Module-06/01-colonnes-df/05-agregations.py
from pyspark.sql.functions import *

meteoDataFrame.columns
meteoDataFrame.printSchema()

meteoDataFrame.select('numer_sta',
        expr('t  - 273.15').alias('temperature'),
        expr('(t + pres/1000)*vv/1000')
             .alias('calc')).show(3)

villes.select('*',expr('altitude * 1000').alias('alt')).show(3)
villes.selectExpr('*','altitude * 1000 as alt').show(3)

meteoDataFrame.selectExpr('numer_sta',
        expr('t  - 273.15').alias('temperature'),
        expr('(t + pres/1000)*vv/1000')
             .alias('calc')).show(3)

meteo.where('id < 8000')\
     .select('annee','mois_jour',
             'temperature')\
     .describe().show()
